lib/CallStack.cpp

```diff
@@ -42,7 +42,7 @@ QoreRWLock* thread_stack_lock;
 QoreRWLock thread_stack_lock;
```

Comment:
why is it pointer on windows? - this way, it won't be initialized and then locking will be in an undefined behavior (no one knows, what value will it have). Also, does it have to be a global variable?

Reply:
I don't remember; unfortunately I did not add a comment for this, but I do remember that it did not work in Windows builds if this was not allocated explicitly.  I don't remember why - or even if I found out why (I suspect I didn't)

Reply:
> this was not allocated explicitly.
but it's not, it's just a pointer pointing to nowhere, isn't it?

Reply:
it's an uninitialized pointer in static memory which is assigned to 0 (`nulllptr`) according the standard (https://stackoverflow.com/questions/3373108/why-are-static-variables-auto-initialized-to-zero).

In Linux, it gets properly constructed in the static construction phase of library initialization, however for some reason on Windows it didn't work at the time, and had to be initialized explicitly.

This was many years ago, so maybe it's not necessary anymore.

Reply:
oh my, it really does exist (static zero initialization, I didn't know that).
but then using it with `QoreAutoRWReadLocker` will do nothing based on
https://github.com/qorelanguage/qore/blob/develop/include/qore/QoreRWLock.h#L126
no locking at all

Reply:
or is it initialized somewhere else?
> had to be initialized explicitly.

Reply:
on windows it's initialized right below the declaration in `init_qore_threads()` before it's used.

Reply:
ah, it's there, ok, it's a bit confusing, ok, so let's consider this comment as resolved then